config/admission/admission.yaml

```diff
@@ -0,0 +1,32 @@
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: "clusterextensions-package-uniqueness.olm.operatorframework.io"
+spec:
+  failurePolicy: Fail
+  paramKind:
+    apiVersion: olm.operatorframework.io/v1alpha1
+    kind: ClusterExtension
+  matchConstraints:
+    resourceRules:
+    - apiGroups:   ["olm.operatorframework.io"]
+      apiVersions: ["v1alpha1"]
+      operations:  ["CREATE", "UPDATE"]
+      resources:   ["clusterextensions"]
+  validations:
+    - expression: "object.spec.packageName != params.spec.packageName"
+      messageExpression: "'Package \"' + string(object.spec.packageName) + '\" is already installed via ClusterExtension \"' +  string(params.metadata.name) + '\"'"
+      reason: Invalid
+
+---
+
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: ValidatingAdmissionPolicyBinding
+metadata:
+  name: "clusterextensions-package-uniqueness-binding.olm.operatorframework.io"
+spec:
+  policyName: "clusterextensions-package-uniqueness.olm.operatorframework.io"
```

Comment:
I'll keep it in draft for now because `kustomize` currently doesn't handle this (even in the latest versions it seems).

I'll report an issue and see what else I can do to make it work

Reply:
Replacing this:
https://github.com/operator-framework/operator-controller/blob/652e4bca63f9e0a04973b5a1ef7ec03ca0cf330b/config/default/kustomization.yaml#L4-L9

With something like this might work:

```yaml
transformers:
- |-
  apiVersion: builtin
  kind: PrefixSuffixTransformer
  metadata:
    name: prefix-all-names
  prefix: operator-controller-
  fieldSpecs:
    - path: metadata/name
    - group: admissionregistration.k8s.io
      kind: ValidatingAdmissionPolicyBinding
      path: spec/policyName
```

Reply:
Reported: https://github.com/kubernetes-sigs/kustomize/issues/5674

Also updated the PR with a workaround.

Reply:
That workaround looks like it worked!

Another way to do this that I've seen is to use `nameReference` in a kustomize configuration file (explained fairly well here: https://www.innoq.com/en/blog/2022/07/advanced-kustomize-features/#supportcrds)

Reply:
Nice! I think `nameReference` looks slightly better. I'll use it instead.